modules/ting_field_search/ting_field_search.module

```diff
@@ -432,39 +433,202 @@ function ting_field_search_ting_search_settings_submit($form, &$form_state) {
+    if ($visibility) {
+      $pages .= "\r\nsearch-blank";
+    }
+
+    // Compare the lowercase internal and lowercase path alias (if any).
+    $page_match = drupal_match_path($path, $pages);
+    if ($path != $_GET['q']) {
+      $page_match = $page_match || drupal_match_path($_GET['q'], $pages);
+    }
+
+    // When $visibility has a value of 0, the profile is displayed on
+    // all pages except those listed in $pages. When set to 1, it
+    // is displayed only on those pages listed in $pages.
+    if (!($visibility xor $page_match)) {
+      $vars['classes_array'][] = 'extended-search-is-open';
+      break;
+    }
   }
+}
 
-  // Dont't alter the form if this is the node search page.
-  if (strpos(current_path(), 'search/node') === 0) {
-    return;
+
+/**
+ * Find the default profile for the current context.
+ *
+ * @return string
+ *   The name of the profile that is default.
+ */
+function ting_field_search_get_default_profile() {
+  // If there's an active profile and it is stick; set it as default value.
+  if ($default_profile = ting_field_search_get_active_profile()) {
+    $default_profile = $default_profile->config['user_interaction']['sticky'] ? $default_profile->name : '';
+  }
+
+  if (empty($default_profile)) {
+    $profiles = ting_field_search_profiles_load_all();
+    foreach ($profiles as $profile) {
+      // Expended search box not checked go to the next profile.
+      if (!array_key_exists('default_profile', $profile->config['user_interaction']) || !$profile->config['user_interaction']['default_profile']) {
+        continue;
+      }
+
+      // If the profile has visibility all always expand the search field.
+      if ($profile->config['user_interaction']['visibility'] === 'all') {
+        $default_profile = $profile->name;
+        break;
+      }
+
+      $visibility = $profile->config['user_interaction']['visibility'] == 'expect' ? 0 : 1;
+      $pages = $profile->config['user_interaction']['pages'];
+      $path = drupal_strtolower(drupal_get_path_alias($_GET['q']));
+
+      // We have an optimization hack to speed up search (by-pass panels 2x
+      // times loading). So all searches are submitted to "/search-blank",
+      // which need to be added to pages to ensure no form validation errors.
+      // If visibility is "expect" it should not be added to the pages list.
+      if ($visibility) {
+        $pages .= "\r\nsearch-blank";
+      }
+
+      // Compare the lowercase internal and lowercase path alias (if any).
+      $page_match = drupal_match_path($path, $pages);
+      if ($path != $_GET['q']) {
+        $page_match = $page_match || drupal_match_path($_GET['q'], $pages);
+      }
+
+      // When $visibility has a value of 0, the profile is displayed on
+      // all pages except those listed in $pages. When set to 1, it
+      // is displayed only on those pages listed in $pages.
+      if (!($visibility xor $page_match)) {
+        $default_profile = $profile->name;
```

Comment:
This seems to be almost the same code as in `ting_field_search_preprocess_html()`. It is also quite complex. Please refactor.

Reply:
This is taken from drupal core, so think this is fine.

Reply:
As far as I can tell [the Block module in Drupal Core where this is taken from has this once](https://github.com/drupal/drupal/blob/7.x/modules/block/block.module#L819-L831) and it does not include an optimization hack requiring 4 lines of comments so I do not think it is the same.

See aakbcms/patches#1 for a suggestion for how to refactor.

Reply:
Changes accepted